store/postgres/tests/store.rs

```diff
@@ -2289,3 +1991,185 @@ fn handle_large_string_with_index() {
         Ok(())
     })
 }
+
+#[derive(Clone)]
+struct WindowQuery(EntityQuery, Arc<DieselStore>);
+
+impl WindowQuery {
+    fn new(store: &Arc<DieselStore>) -> Self {
+        WindowQuery(
+            user_query()
+                .filter(EntityFilter::GreaterThan("age".into(), Value::from(0)))
+                .first(10),
+            store.clone(),
+        )
+        .default_window()
+    }
+
+    fn default_window(mut self) -> Self {
+        let entity_types = match self.0.collection {
+            EntityCollection::All(entity_types) => entity_types,
+            EntityCollection::Window(_) => {
+                unreachable!("we do not use this method with a windowed collection")
+            }
+        };
+        let windows = entity_types
+            .into_iter()
+            .map(|child_type| {
+                let attribute = WindowAttribute::Scalar("favorite_color".to_owned());
+                let link = EntityLink::Direct(attribute);
```

Comment:
Are we indirectly testing the other `EntityLink` variant or should we add tests for this here as well?

Reply:
Yes, they are covered by the tests in `core/tests/interfaces.rs` and `graphql/tests/query.rs`

Reply:
Ah, ok, good.